tests/integration-tests/tests/networking/test_cluster_networking.py

```diff
@@ -80,67 +86,216 @@ def test_existing_eip(existing_eip, pcluster_config_reader, clusters_factory):
     connection.run("cat /var/log/cfn-init.log", timeout=60)
 
 
-def _test_fsx_in_private_subnet(cluster, os, region, scheduler, fsx_mount_dir, bastion_factory):
+def _test_fsx_in_private_subnet(cluster, os, region, scheduler, fsx_mount_dir, bastion_instance):
     """Test FSx can be mounted in private subnet."""
-    bastion_ip = bastion_factory()
     logging.info("Sleeping for 60 sec to wait for bastion ssh to become ready.")
     time.sleep(60)
-    logging.info(f"Bastion_ip: {bastion_ip}")
-    remote_command_executor = RemoteCommandExecutor(cluster, bastion=f"ec2-user@{bastion_ip}")
+    logging.info(f"Bastion: {bastion_instance}")
+    remote_command_executor = RemoteCommandExecutor(cluster, bastion=bastion_instance)
     scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
     fsx_fs_id = get_fsx_fs_id(cluster, region)
     assert_fsx_lustre_correctly_mounted(remote_command_executor, fsx_mount_dir, os, region, fsx_fs_id)
     assert_fsx_lustre_correctly_shared(scheduler_commands, remote_command_executor, fsx_mount_dir)
 
 
+@pytest.mark.usefixtures("enable_vpc_endpoints")
+@pytest.mark.usefixtures("instance")
+def test_cluster_in_no_internet_subnet(
```

Comment:
given a compute node shall we assert that:
1. Running a scontrol show node {nodename} gives you the expected NodeName and NodeHostname?
2. the hostname of the compute node is configured as expected (A quick way to retrieve it is to run `srun hostname -w nodename` or `ssh nodehostname hostname`)